pkg/credentials/service_account_credentials.go

```diff
@@ -62,6 +63,58 @@ func NewCredentialBulder(client client.Client, config *v1.ConfigMap) *Credential
 	}
 }
 
+func (c *CredentialBuilder) CreateStorageSpecSecretVolumeAndEnv(namespace string, storageKey string,
+	storageSecretName string, overrideParams map[string]string, container *v1.Container, volumes *[]v1.Volume) error {
+	secret := &v1.Secret{}
```

Comment:
As mentioned in other comment I think we want to change the logic here a bit:

If secret key is non-empty the look it up, otherwise get the type from `overrideParams` and based on that look up a secret key using that type name, like `s3_default`. If the secret is not found, we should only fail in the former case (where the secretKey was specified explicitly), otherwise, just make the overrideParams be the actual params.

Reply:
I updated with your new logic, please take a look

Reply:
@Tomcli it doesn't quite cover everything that I was suggesting... might be easier for me to just push a small commit, will aim to do that soon. But most of this logic will be moving to the storage initializer anyhow right?

Reply:
I kept the secret checking logics inside the golang code. If the secret is found, it will pass the secret value as a new env variable to the pod. It also passes all the `overrideParams` as another env variable to the pod if exist.